### ISSUE TITLE
Updating from 3.8.0 to 3.14.0 breaks due to containsSequence

### ISSUE THREAD

**Issue by SElab2019** [joel-costigliola/assertj-core#1756]
When I try to upgrade assertj-core from 3.8.0 to the latest 3.14.0. The following code breaks. The issue starts from version 3.9.0. The original test case is complicated, so I just short the string into a  text file.
```
import static org.assertj.core.api.Assertions.assertThat;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

import org.junit.Test;

public class Testassertj {
	
	@Test
	public void demo() throws IOException {
		
		String content = new String ( Files.readAllBytes( Paths.get("/test.txt") ) );
		assertThat(content).containsSequence("msg6", "msg7", "msg1", "msg2", "msg4", "msg3", "msg5");
	}
}
```
The text file content is:
```
SonarQube analysis reported 7 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] foo: msg6 [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://myserver/coding_rules#rule_key=repo%3Arule)
1. ![BLOCKER][BLOCKER] foo:src/Foo.php: msg7 [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://myserver/coding_rules#rule_key=repo%3Arule)
1. ![BLOCKER][BLOCKER] foo:src/Foo.php: msg1 [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://myserver/coding_rules#rule_key=repo%3Arule)
1. ![BLOCKER][BLOCKER] foo:src/Foo2.php: msg2 [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://myserver/coding_rules#rule_key=repo%3Arule)
1. ![CRITICAL][CRITICAL] foo:src/Foo.php: msg4 [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://myserver/coding_rules#rule_key=repo%3Arule)
1. ![MAJOR][MAJOR] foo:src/Foo.php: msg3 [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://myserver/coding_rules#rule_key=repo%3Arule)
1. ![INFO][INFO] foo:src/Foo.php: msg5 [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://myserver/coding_rules#rule_key=repo%3Arule)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
```

The test should pass, but it failed with an error:
```
Testassertj
demo(Testassertj)
java.lang.AssertionError: 
Expecting:
 <"SonarQube analysis reported 7 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] foo: msg6 [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://myserver/coding_rules#rule_key=repo%3Arule)
1. ![BLOCKER][BLOCKER] foo:src/Foo.php: msg7 [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://myserver/coding_rules#rule_key=repo%3Arule)
1. ![BLOCKER][BLOCKER] foo:src/Foo.php: msg1 [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://myserver/coding_rules#rule_key=repo%3Arule)
1. ![BLOCKER][BLOCKER] foo:src/Foo2.php: msg2 [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://myserver/coding_rules#rule_key=repo%3Arule)
1. ![CRITICAL][CRITICAL] foo:src/Foo.php: msg4 [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://myserver/coding_rules#rule_key=repo%3Arule)
1. ![MAJOR][MAJOR] foo:src/Foo.php: msg3 [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://myserver/coding_rules#rule_key=repo%3Arule)
1. ![INFO][INFO] foo:src/Foo.php: msg5 [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://myserver/coding_rules#rule_key=repo%3Arule)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
">
to contain sequence:
 <["msg6", "msg7", "msg1", "msg2", "msg4", "msg3", "msg5"]>

	at Testassertj.demo(Testassertj.java:17)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:89)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:41)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:541)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:763)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:463)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:209)
```

**Comment by joel-costigliola** [comment]
reason is a breaking change in 3.9.0 (which supersedes 2.9.0 with java 8 specific feature), details here https://joel-costigliola.github.io/assertj/assertj-core-older-releases.html#assertj-core-2.9.0.

javadoc extract:
```
public SELF containsSequence​(Iterable<? extends CharSequence> values)

Verifies that the actual CharSequence contains all the values of the given Iterable in the Iterable iteration order without any other values between them.

Breaking change since 2.9.0: in previous versions this assertion behaved like containsSubsequence and allowed other values between the sequence values. 
```

TL;DR replace by `containsSubsequence`

**Comment by SElab2019** [comment]
Hi @joel-costigliola , thanks so much for your help. Actually, I do try to find the breaking change in the release note. I found the issue starts from 3.9.0, and I didn't find any announcement there. BTW, would you mind also adding one line in the 3.9.0 release note?

**Comment by joel-costigliola** [comment]
the 3.9.0 notes has a link to the 2.9.0 ones, not super handy but avoid duplication.
what I need to do is add a link on the new site to the older release notes

**Comment by SElab2019** [comment]
Alright. That's awesome. Thanks again for your help.
